graphql_compiler/compiler/ir_lowering_match/optional_traversal.py

```diff
@@ -559,3 +560,13 @@ def lower_context_field_expressions(compound_match_query):
             )
 
     return CompoundMatchQuery(match_queries=new_match_queries)
+
+
+def lower_sub_queries(compound_match_query):
```

Comment:
Either this method needs to not be generic (probably preferred option), or it needs to not be in the `optional_traversal.py` file. I think it's better to not have this "generically lowering sub-queries" approach, and instead focus on one-thing-per-lowering-step functions.

So I'd suggest renaming this function to something less generic and more directly related to what it does.

Reply:
Wouldn't that mean writing this same loop structure multiple times for each lowering step we want to apply to the `CompoundMatchQuery`?
I thought having it be generic would allow us to add more lowering steps into the loop. In that case, this function could maybe go in `__init__.py`?